js_modules/dagster-ui/packages/ui-core/src/asset-graph/useAssetGraphData.tsx

```diff
@@ -38,14 +40,19 @@ export type AssetGraphQueryItem = GraphQueryItem & {
  * uses this option to implement the "3 of 4 repositories" picker.
  */
 export function useAssetGraphData(opsQuery: string, options: AssetGraphFetchScope) {
-  const fetchResult = useQuery<AssetGraphQuery, AssetGraphQueryVariables>(ASSET_GRAPH_QUERY, {
-    notifyOnNetworkStatusChange: true,
+  const fetchResult = useIndexedDBCachedQuery<AssetGraphQuery, AssetGraphQueryVariables>({
+    query: ASSET_GRAPH_QUERY,
     variables: {
       pipelineSelector: options.pipelineSelector,
       groupSelector: options.groupSelector,
     },
+    key: JSON.stringify({
+      pipelineSelector: options.pipelineSelector,
+      groupSelector: options.groupSelector,
+    }),
+    version: 1,
```

Comment:
This feels like it's kinda pushing it in terms of key variance... These indexeddb caches are never cleaned up and this seems close to the "it'll just fill your Chrome Application Support folder with new keys forever" threshold, since pipeline and group are somewhat combinatorial.

Now that we're using this all over the app, I'd really like to see if we can switch this to use the Cache APIs - you can read/write data to a key pretty easily there too, it's not limited to 5MB, and it's easier for users to clear on their own. Then I'd feel comfortable making new keys forever 😅 

Definitely something we can punt to a follow-up though

Reply:
I'm planning on adding an option to let you delete all of the indexeddbs from the user preferences setting